File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    
    const countList = {};
    
    for (let i = 0; i < tangerine.length; i++) {
        countList[tangerine[i]] = (countList[tangerine[i]] || 0) + 1;
    }
    
    const sortedArr = Object.values(countList).sort((a, b) => b- a);
    
    while(k > 0) {
        k -= sortedArr.shift();
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int t: tangerine) {
            map.put(t, map.getOrDefault(t, 0) + 1);
        }
        
        int[] sorted = map.values()
            .stream()
            .sorted(Collections.reverseOrder())
            .mapToInt(Integer::intValue)
            .toArray();
        
        int answer = 0;
        for (int i: sorted) {
            if (k <= 0) break;
            k -= i;
            answer++;
        }
        
        return answer;
    }
}